app/services/facebook_authenticator.rb

```diff
@@ -12,10 +12,14 @@ def initialize(auth_code)
   end
 
   def authenticate!
+    access_token  = get_access_token(@auth_code)
```

Comment:
Point was to improve clarity of the steps taken in facebook auth. Yesterday we struggled to see where things were happening because they were too implicit. Having a explicit get_access_token that gets passed `@auth_code` improves it, I think. It's an instance variable so it can be assigned no initialization and available to `authenticate!`. Removing clarity because of a strict of a "rule" seems silly imo. But it's your call